integration_tests/auction_test.go

```diff
@@ -0,0 +1,256 @@
+package integration_tests
+
+import (
+	"context"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/peggyjv/sommelier/v4/x/auction/types"
+	cellarfees "github.com/peggyjv/sommelier/v4/x/cellarfees/types"
+)
+
+func (s *IntegrationTestSuite) TestAuctionModule() {
+	s.Run("Bring up chain, test governance proposal to set token prices, submit some bids, and finish an auction", func() {
+		val := s.chain.validators[0]
+		kb, err := val.keyring()
+		s.Require().NoError(err)
+		val0ClientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", val.keyInfo.GetAddress())
+		s.Require().NoError(err)
+		auctionQueryClient := types.NewQueryClient(val0ClientCtx)
+
+		// Verify auction created for testing exists
+		auctionQuery := types.QueryActiveAuctionRequest{
+			AuctionId: uint32(1),
+		}
+		s.T().Log("Verifying expected auction exists ...")
+		auctionResponse, err := auctionQueryClient.QueryActiveAuction(context.Background(), &auctionQuery)
+		s.Require().NoError(err)
+		s.Require().Equal(auctionResponse.Auction.Id, uint32(1))
+		s.T().Log("Expected auction found!")
+
+		s.T().Logf("Create governance proposal to update some token prices")
+		orch := s.chain.orchestrators[0]
+		orchClientCtx, err := s.chain.clientContext("tcp://localhost:26657", orch.keyring, "orch", orch.keyInfo.GetAddress())
+		s.Require().NoError(err)
+		orch0Address := s.chain.orchestrators[0].keyInfo.GetAddress().String()
+
+		proposal := types.SetTokenPricesProposal{
+			Title:       "initial token price submission",
+			Description: "our first token prices",
+			TokenPrices: []*types.ProposedTokenPrice{
+				{
+					Denom:    "gravity0x3506424f91fd33084466f402d5d97f05f8e3b4af",
+					UsdPrice: sdk.MustNewDecFromStr("1.0"),
+				},
+				{
+					Denom:    "gravity0x5a98fcbea516cf06857215779fd812ca3bef1b32",
+					UsdPrice: sdk.MustNewDecFromStr("0.25"),
+				},
+				{
+					Denom:    types.UsommDenom,
+					UsdPrice: sdk.MustNewDecFromStr("0.5"),
+				},
+			},
+		}
+
+		proposalMsg, err := govtypes.NewMsgSubmitProposal(
+			&proposal,
+			sdk.Coins{
+				{
+					Denom:  testDenom,
+					Amount: stakeAmount.Quo(sdk.NewInt(2)),
```

Comment:
The test genesis should specify an exact deposit amount that is necessary rather than depositing half of your stake (also, where does this variable come from? I don't see it initialized...)

Reply:
Yeah I just followed the other integration tests as a template here (e.g. https://github.com/PeggyJV/sommelier/blob/main/integration_tests/cork_test.go#L152-L161)

The stakeAmount is set in the setup file here https://github.com/PeggyJV/sommelier/blob/main/integration_tests/setup_test.go#L50, If I'm reading the minimum deposit amount correctly, I think it's just 1 unit so this value probably doesnt matter too much in the test? https://github.com/PeggyJV/sommelier/blob/main/integration_tests/setup_test.go#L287